x-pack/plugins/apm/server/lib/helpers/create_es_client/create_apm_event_client/index.ts

```diff
@@ -193,12 +199,63 @@ export class APMEventClient {
     });
   }
 
+  async msearch<TParams extends APMEventESSearchRequest>(
+    operationName: string,
+    ...allParams: TParams[]
+  ): Promise<TypedMSearchResponse<TParams>> {
+    const searches = allParams
+      .map((params) => {
+        const { index, filters } = getRequestBase({
+          apm: params.apm,
+          indices: this.indices,
+        });
+
+        const searchParams: [MsearchMultisearchHeader, MsearchMultisearchBody] =
```

Comment:
nit: Perhaps return this directly and then rename `searches` to `searchParams`

Reply:
I'm using an intermediate variable here for more accurate typings. What the client expects is `Array<MsearchMultisearchHeader  | MsearchMultisearchBody>`, but it actually is something like `[MsearchMultisearchHeader, MsearchMultisearchBody, MsearchMultisearchHeader, MsearchMultisearchBody, ... ]` which is hard to type.